Expensify/App
# Update react-navigation to latest versions

### Details
There is a bug in the react navigation library that has already been fixed and we just need to update.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/178594

### Tests

1. `npm i && npm run web`
2. Sign out of NewDot
2. Sign up for a new account in OldDot on a mobile web (small width on Chrome or iPhone user agent in safari or iOS simulator)
3. Tap the first link
4. Verify you are brought to the workspace page
5. Verify you do not see a white screen

### QA Steps
1. Sign out of NewDot
2. Sign up for a new account in OldDot on a mobile web (use actual device)
3. Tap the first link
4. Verify you are brought to the workspace page
5. Verify you do not see a white screen
6. Repeat 10+ times to verify the white screen can't happen

### Tested On


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
